### ISSUE TITLE
toLocaleString documentation is ambigious

### ISSUE THREAD

**Issue by heatherbooker** [mdn/content#30436]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString

### What specific section or headline is this issue about?

toLocaleString's introduction

### What information was incorrect, unhelpful, or incomplete?

The page says 
> When formatting large numbers of numbers, it is better to create a [Intl.NumberFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat) object and use the function provided by its [format()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/format) method.

"large numbers of numbers" can be ambiguous - with my colleague, one of us understood "1,000,000s of separate numbers" while the other understood "one number 1,000,000".

### What did you expect to see?

That section could be clearer (like simply changing it to "when formatting many many numbers", and or adding: what is a large number of numbers? 100s? 1,000,000,000s?).

I think it would also help to explain _why_ it is better to use NumberFormat - that would also help disambiguate, and would help users decide if "better" is meaningful to them.

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/number/tolocalestring`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/number/tolocalestring/index.md
* Last commit: https://github.com/mdn/content/commit/3cfd663738e9963157d90f359789d675a6662ec2
* Document last modified: 2023-08-09T10:45:44.000Z

</details>

**Comment by Josh-Cena** [comment]
> with my colleague, one of us understood "1,000,000s of separate numbers" while the other understood "one number 1,000,000".

Both are fine. The only thing that matters is how many times you call `toLocaleString`. Whether each time the formatted number is the same is \*probably\* irrelevant, unless the engine does specific caching for each return value. Of course you can also cache the result yourself, but that's yet another topic.

The idea is this: the localization strings are like a giant database. `toLocaleString` is doing a search within the entire database every time, which is inefficient. `Intl.NumberFormat` first pulls out a specific "slice" of the database (based on the locale and options) and then the scope for future searches will be much smaller. This optimization is on a constant level (citation needed) and constructing a formatter instance does come with its own overhead and ergonomics tradeoffs, so for small numbers of `toLocaleString` calls it probably doesn't matter. Nothing in performance has a carved-in-stone threshold that "if you do it more than x times, option A is always going to be more superior"; rather, it's an _asymptotic_ analysis. It also depends a lot on the implementation, because again the optimization is only on a constant level.

Given all this, how would you suggest us rephrase it to make it less ambiguous?

**Comment by edwardloveall** [comment]
This looks great. I didn't know any of this so thank you for the explanation @Josh-Cena. I wouldn't change anything.
